packages/components/data-source/src/fetch-data-source.ts

```diff
@@ -0,0 +1,157 @@
+import { DataSource, type DataSourceSort } from './data-source.js';
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+export interface FetchDataSourceCallbackOptions<T = any> {
```

Comment:
Maybe `unknown` instead of `any` would be better?

Reply:
If you use unknown, then you will get warnings/errors in code that uses it, without a type (speaking from experience). So while `unknown` is more correct, `any` is more practical.